### ISSUE TITLE
[Err] 1690 - constant -6884244073709551616 overflows bigint

### ISSUE THREAD

**Issue by GiveMeSunShine** [pingcap/tidb#11606]
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do?
If possible, provide a recipe for reproducing the error.
I create table with DDL SQL：
> DROP TABLE IF EXISTS `eth_analysis_address`;
CREATE TABLE `eth_analysis_address` (
  `address` char(42) NOT NULL COMMENT '地址',
  `balance` decimal(60,0) NOT NULL DEFAULT '0' COMMENT '当前余额',
  `income` decimal(60,0) NOT NULL DEFAULT '0' COMMENT '总收入',
  `inblocknum` bigint(20) NOT NULL DEFAULT '0' COMMENT '首次出现的块号',
  `modblocknum` bigint(20) NOT NULL DEFAULT '0' COMMENT '最近更新的块号',
  `modcount` bigint(20) NOT NULL DEFAULT '0' COMMENT '参与交易次数',
  `type` tinyint(3) UNSIGNED ZEROFILL NOT NULL DEFAULT '0' COMMENT '0:普通地址 1：合约地址',
  PRIMARY KEY (`address`),
  KEY `balance_index` (`balance`),
  KEY `inblocknum_index` (`inblocknum`),
  KEY `modblocknum_index` (`modblocknum`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_bin;

When I executed SQL twice with insert SQL, it reported an error.
insert SQL:
> insert into eth_analysis_address(address, balance, income, inblocknum ,modblocknum,modcount,type) values ('0xbb7b8287f3f0a933474a79eae42cbca977791171', 11562500000000000000,11562500000000000000, 97, 97,1,0) on duplicate key update balance =balance+11562500000000000000,modblocknum=97,income=income+11562500000000000000,modcount=modcount+1

reported error:
> [Err] 1690 - constant -6884244073709551616 overflows bigint

After testing, it was found that it was a decimal type field report error, but this should not be an error.

2. What did you expect to see?
The SQL should not report errors, Table field balance and income should not have an overflow exception。

3. What did you see instead?
reported error:
> [Err] 1690 - constant -6884244073709551616 overflows bigint

4. What version of TiDB are you using (`tidb-server -V` or run `select tidb_version();` on TiDB)?
tidb-server -V
> Release Version: v2.1.13
Git Commit Hash: 6b5b1a6802f9b8f5a22d8aab24ac80729331e1bc
Git Branch: HEAD
UTC Build Time: 2019-06-21 12:27:08
GoVersion: go version go1.12 linux/amd64
Race Enabled: false
TiKV Min Version: 2.1.0-alpha.1-ff3dd160846b7d1aed9079c389fc188f7f5ea13e
Check Table Before Drop: false

**Comment by zz-jason** [comment]
@SunRunAway PTAL

**Comment by GiveMeSunShine** [comment]
@zz-jason Will this bug be fixed in which version?

**Comment by zz-jason** [comment]
We'll fix it soon and cherry pick it to the next release version.

**Comment by GiveMeSunShine** [comment]
@zz-jason Why is this bug not fixed in version 2.1.17?Will this bug be fixed in which version?

**Comment by nullnotnil** [comment]
Confirming that this issue has been fixed in master. I am going to close this issue now, but please feel free to re-open if you have additional questions. Thanks!

```sql
DROP TABLE IF EXISTS eth_analysis_address;
CREATE TABLE eth_analysis_address (
address char(42) NOT NULL COMMENT '地址',
balance decimal(60,0) NOT NULL DEFAULT '0' COMMENT '当前余额',
income decimal(60,0) NOT NULL DEFAULT '0' COMMENT '总收入',
inblocknum bigint(20) NOT NULL DEFAULT '0' COMMENT '首次出现的块号',
modblocknum bigint(20) NOT NULL DEFAULT '0' COMMENT '最近更新的块号',
modcount bigint(20) NOT NULL DEFAULT '0' COMMENT '参与交易次数',
type tinyint(3) UNSIGNED ZEROFILL NOT NULL DEFAULT '0' COMMENT '0:普通地址 1：合约地址',
PRIMARY KEY (address),
KEY balance_index (balance),
KEY inblocknum_index (inblocknum),
KEY modblocknum_index (modblocknum)
) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_bin;

insert into eth_analysis_address(address, balance, income, inblocknum ,modblocknum,modcount,type) values ('0xbb7b8287f3f0a933474a79eae42cbca977791171', 11562500000000000000,11562500000000000000, 97, 97,1,0) on duplicate key update balance =balance+11562500000000000000,modblocknum=97,income=income+11562500000000000000,modcount=modcount+1;

insert into eth_analysis_address(address, balance, income, inblocknum ,modblocknum,modcount,type) values ('0xbb7b8287f3f0a933474a79eae42cbca977791171', 11562500000000000000,11562500000000000000, 97, 97,1,0) on duplicate key update balance =balance+11562500000000000000,modblocknum=97,income=income+11562500000000000000,modcount=modcount+1;

..

mysql> insert into eth_analysis_address(address, balance, income, inblocknum ,modblocknum,modcount,type) values ('0xbb7b8287f3f0a933474a79eae42cbca977791171', 11562500000000000000,11562500000000000000, 97, 97,1,0) on duplicate key update balance =balance+11562500000000000000,modblocknum=97,income=income+11562500000000000000,modcount=modcount+1;
Query OK, 1 row affected (0.02 sec)

mysql> 
mysql> insert into eth_analysis_address(address, balance, income, inblocknum ,modblocknum,modcount,type) values ('0xbb7b8287f3f0a933474a79eae42cbca977791171', 11562500000000000000,11562500000000000000, 97, 97,1,0) on duplicate key update balance =balance+11562500000000000000,modblocknum=97,income=income+11562500000000000000,modcount=modcount+1;
Query OK, 2 rows affected (0.01 sec)

mysql> 
mysql> select tidb_version()\G
*************************** 1. row ***************************
tidb_version(): Release Version: v4.0.0-beta.2-771-gca41972fb
Edition: Community
Git Commit Hash: ca41972fbac068c8a5de107d9075f09ac68842ac
Git Branch: master
UTC Build Time: 2020-07-14 02:41:21
GoVersion: go1.13
Race Enabled: false
TiKV Min Version: v3.0.0-60965b006877ca7234adaced7890d7b029ed1306
Check Table Before Drop: false
1 row in set (0.00 sec)
```
